### PR TITLE
Update of muon regional iterative tracking in HI reco

### PR DESCRIPTION
**Pull request by echapon**
The muon regional iterative (regit) tracking in heavy-ion reconstruction is updated with the following changes:
- removing the LowPt, Detached and TobTec iterations. These iterations don't change the muon reconstruction efficiency, so removing them saves a bit of time in reconstruction.
- fixing the algorithm name assigned to tracks found by muon regit, with an existing name. The tracks are assigned the flag "undefAlgorithm", this can be changed if someone has a strong opinion about it.

Documentation of the effect of these changes on the reconstruction performance and resources needed is under finalization and made available very soon. As stated above, the HI muon reconstruction is basically unchanged, only a little bit faster.

### CODE REVIEW

**Comment by slava77** [comment]
@rovere @VinInn are there any issues with setting the tracking algorithm to 0/undefined (and have multiple algorithms running)?

**Comment by VinInn** [comment]
besides debug and validation algo id is used only as last resort in ListMerging
algo id SHALL NOT be used in any track classification

**Comment by slava77** [comment]
@cmsbuild please test

File: RecoHI/HiMuonAlgos/python/HiRegitMuonDetachedTripletStep_cff.py [hunk 1]
```diff
@@ -63,7 +63,7 @@
 
 # fitting: feed new-names
 hiRegitMuDetachedTripletStepTracks                 = RecoTracker.IterativeTracking.DetachedTripletStep_cff.detachedTripletStepTracks.clone(
-    AlgorithmName = cms.string('iter6'),
+    AlgorithmName = cms.string('undefAlgorithm'),
```
Comment (VinInn) [inline RecoHI/HiMuonAlgos/python/HiRegitMuonDetachedTripletStep_cff.py]: are you sure?
would not be better to use the official naming scheme?

Comment (mandrenguyen) [inline RecoHI/HiMuonAlgos/python/HiRegitMuonDetachedTripletStep_cff.py]: Well, the muon regional tracks get merged with the standard global tracks.  We thought maybe the naming scheme should keep them independent, such that customers of standard tracking can tell them apart.  On the other hand, we don't set the 'tight' and 'highPurity' bits for the muon regional tracks, which should be used for any analysis.  If you think it's better that we use the standard naming convention we'll make the change.

**Comment by slava77** [comment]
@cmsbuild please test
the jenkins tests made earlier in CMSSW_7_5_X_2015-05-11-2300 were not that useful due to failing HI matrix workflow in the IB itself

**Comment by slava77** [comment]
@VinInn @rovere  @echapon  @mandrenguyen 
would it make sense to add several somewhat non-specific iterations, e.g., regionalIter0 ... regionalIter4, regionalIterX to be used for this and similar purposes ?

**Comment by echapon** [comment]
Yes, that's a possibility I thought of, which is easy to implement if people like it (it's just about adding algorithm names in TrackBase.h right?). I will implement it if the tracking POG likes the idea.

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/9049	10ba8bc
- changes in the configs are as intended (although additional fine-tuning that requires a new set of track algo enums and labels defined may happen in a following PR).
- changes affect only HI workflows
- tested locally in CMSSW_7_5_0_pre4 /local area sign551/
    - default matrix: 140.53 (data) and 140.0 (MC)
    - dimuon skim using 140.53 config and /store/results/hin/StoreResults/HIDiMuon/USER/L2DoubleMu3Skim_v10_38dff9fa051006d6e895e3c1df676d76-v1/40000/001FA005-07D0-E411-A6C2-0026B95A4D42.root
- most visible changes are in the technical performance in the CPU: about 0.8% reduction from removal of lowPt triplet, detached tripled, and tob-tec  mu regional iterations
- event content changes very little: 
    - 0.2% to 0.4% less hiGeneralTracks (higher reduction in the dimu skim)
    - 0.9% less reco muons in plain 140.53 and 1.2% less in the dimu skim
- physics performance changes are pretty minor based on the monitored quantities (in particular, checking the MC sample validation plots: efficiencies and fake rates are practically the same on the "global" scale)

**Comment by davidlange6** [comment]
+1

**Pull request merged**